dvc/dvcfile.py

```diff
@@ -289,6 +289,33 @@ def _dump_pipeline_file(self, stage):
 
         self.repo.scm_context.track_file(self.relpath)
 
+    def _dump_pipeline_file_artifacts(self, name, artifact):
+        # I can either generalize `_dump_pipeline_file` or implement a separate one
+        # happens that `_dump_pipeline_file` is very much tailored to the `Stage` class
+        # WDYT - does it worth to generalize?
```

Comment:
This looks pretty ugly to me. Let's make a conscious decision here: either we make this OK here, or I simply move this logic to Studio. I would really like to keep it in DVC since it's both easier to use and change. But if that's will be very tough to do and it won't worth it, moving it to Studio is fine. @skshetry can you please help me understand how this can be done better if we keep it in DVC?

Reply:
> I would really like to keep it in DVC
➕

Reply:
Better to move it to `Artifacts` for now.
Also, please decouple "removal" of artifacts from update/add.
And let's skip logging messages for now, as dvc cli is not going to use it.